File: cfgs/java_cfgs_foo/SWAP_BITS_IN_A_GIVEN_NUMBER.java
Nodes
1;special;;;
2;statement;int set1 = (foo_1 >> foo_2) & ((1 << foo_3) - 1);14;
3;statement;int set2 = (foo_1 >> foo_2) & ((1 << foo_3) - 1);15;
4;statement;int xor = foo_1 ^ foo_2;16;
5;statement;xor = (foo_1 << foo_2) | (foo_3 << foo_4);17;
6;statement;int result = foo_1 ^ foo_2;18;
7;statement;return foo_1;19;
8;exit;;;
Edges
0;2;
1;3;
2;4;
3;5;
4;6;
5;7;
6;8;
7;

File: cfgs/pythoncfgs/SWAP_BITS_IN_A_GIVEN_NUMBER.py
Nodes
1;statement;foo_1 = foo_2 >> foo_3 & (1 << foo_4) - 1;8;
2;statement;foo_1 = foo_2 >> foo_3 & (1 << foo_4) - 1;9;
3;statement;foo_1 = foo_2 ^ foo_3;10;
4;statement;foo_1 = foo_2 << foo_3 | foo_4 << foo_5;11;
5;statement;foo_1 = foo_2 ^ foo_3;12;
6;statement;return foo_1;13;
7;exit;;
Edges
0;2;
1;3;
2;4;
3;5;
4;6;
5;7;
6;
